### DISCUSSION TITLE
How to get the ephemeral port used by the debugger before debug session starts?

### DISCUSSION THREAD

**Discussion by infiniteregrets** [microsoft/vscode-discussions#499] [Extension Development QnA] [answered]
My use case needs to know what the port being used by the debug session is before the session starts, so that I can ignore it. Would appreciate any help

**Comment by isidorn** [comment, marked as answer]
You can read the launch.json in case the port is included there.
Not sure if there is any better way to do it. It is not exposed directly in the API.

**Reply by infiniteregrets** [reply]
is there a chance it would be exposed in the API? it would simplify our implementation

**Reply by isidorn** [reply]
No current plans for this.
